### PR TITLE
[feature](cloud) Add priority network support for meta-service registry

### PR DESCRIPTION
**Pull request by gavinchou**
Meta-service registry IP may be 127.0.0.1 without priority network support. It will be a problem for syncing if there are multiple meta-service.
This PR introduces CIDR-based filter to select a true IP for meta-service when registering itself.

### CODE REVIEW

**Comment by gavinchou** [comment]
run buildall

**Comment by gavinchou** [comment]
run buildall

**Review by dataroaring — APPROVED** [review]
LGTM

File: cloud/src/common/network_util.cpp [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+    localhost_str = "";
+    for (auto addr_it = hosts.begin(); addr_it != hosts.end(); ++addr_it) {
+        if ((*addr_it).is_address_v4()) {
+            VLOG_DEBUG << "check ip=" << addr_it->get_host_address_v4();
+            if ((*addr_it).is_loopback_v4()) {
+                loopback = addr_it->get_host_address_v4();
+            } else if (!priority_cidrs.empty()) {
+                if (is_in_prior_network(addr_it->get_host_address_v4())) {
+                    localhost_str = addr_it->get_host_address_v4();
+
```
Comment (w41ter) [inline cloud/src/common/network_util.cpp:210]:
```suggestion
```

File: cloud/src/common/network_util.cpp [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+    std::string loopback;
+    localhost_str = "";
+    for (auto addr_it = hosts.begin(); addr_it != hosts.end(); ++addr_it) {
+        if ((*addr_it).is_address_v4()) {
+            VLOG_DEBUG << "check ip=" << addr_it->get_host_address_v4();
+            if ((*addr_it).is_loopback_v4()) {
+                loopback = addr_it->get_host_address_v4();
+            } else if (!priority_cidrs.empty()) {
+                if (is_in_prior_network(addr_it->get_host_address_v4())) {
```
Comment (w41ter) [inline cloud/src/common/network_util.cpp:208]:
```suggestion
            } else if (!priority_cidrs.empty() && is_in_prior_network(addr_it->get_host_address_v4())) {
```

**Review by w41ter — APPROVED** [review]

**Pull request merged**